xblock/utils/resources.py

```diff
@@ -0,0 +1,101 @@
+"""
+Helper class (ResourceLoader) for loading resources used by an XBlock
+"""
+
+import os
+import sys
+import warnings
+
+import pkg_resources
+from django.template import Context, Template, Engine
+from django.template.backends.django import get_installed_libraries
+from mako.lookup import TemplateLookup as MakoTemplateLookup
+from mako.template import Template as MakoTemplate
+
+
+class ResourceLoader:
+    """Loads resources relative to the module named by the module_name parameter."""
+    def __init__(self, module_name):
+        self.module_name = module_name
+
+    def load_unicode(self, resource_path):
+        """
+        Gets the content of a resource
+        """
+        resource_content = pkg_resources.resource_string(self.module_name, resource_path)
+        return resource_content.decode('utf-8')
+
+    def render_django_template(self, template_path, context=None, i18n_service=None):
+        """
+        Evaluate a django template by resource path, applying the provided context.
+        """
+        context = context or {}
+        context['_i18n_service'] = i18n_service
+        libraries = {
+            'i18n': 'xblock.utils.templatetags.i18n',
+        }
+
+        installed_libraries = get_installed_libraries()
+        installed_libraries.update(libraries)
+        engine = Engine(libraries=installed_libraries)
+
+        template_str = self.load_unicode(template_path)
+        template = Template(template_str, engine=engine)
+        rendered = template.render(Context(context))
+
+        return rendered
+
+    def render_mako_template(self, template_path, context=None):
+        """
+        Evaluate a mako template by resource path, applying the provided context
```

Comment:
Nit: It would be good to mark `render_mako_template` as deprecated too. Django templates or React UI is much preferable to Mako.

Reply:
@bradenmacdonald done

Reply:
Thanks :)